src/PhotonBackground.cpp

```diff
@@ -63,64 +66,377 @@ static PhotonFieldScaling scalingStecker16_upper("IRB_Stecker16_upper");
+        ss2 = 74.1*std::pow(x, -0.44) + 62.*std::pow(s, 0.08);
+        double cs_tmp = 0.96*(1.-std::exp(-ss1))*ss2;
+        cross_diffr1 = 0.14*cs_tmp;
+        cross_diffr2 = 0.013*cs_tmp;
+        double cs_delta = cross_frag2
+                        - (cross_diffr1+cross_diffr2-cross_diffr);
+        if (cs_delta < 0.) {
+            cross_frag2 = 0.;
+            cs_multi += cs_delta;
+        } else {
+            cross_frag2 = cs_delta;
+        }
+        cross_diffr = cross_diffr1 + cross_diffr2;
+        cs_multidiff = cs_multi + cross_diffr;
+    // *****************************************
+    } else {
+        cross_diffr = 0.;
+        cross_diffr1 = 0.;
+        cross_diffr2 = 0.;
+        cs_multidiff = 0.;
+        cs_multi = 0.;
+    }
+    return cross_res+cross_dir+cs_multidiff+cross_frag2;
+}  // end crossection
+
+
+double Photon_Field::Pl(double x, double xth, double xmax, double alpha) const {
+/*
+    - input: photon energy [eV], threshold [eV], max [eV], unknown [no unit]
+    - output: unknown [no unit]
+    - called by: crossection
+*/
+    if (xth > x)
+        return 0.;
+    double a = alpha*xmax/xth;
+    double prod1 = std::pow((x-xth)/(xmax-xth), (a-alpha));
+    double prod2 = std::pow(x/xmax, -a);
+    return prod1*prod2;
+}  // end Pl
+
+
+double Photon_Field::Ef(double x, double th, double w) const {
+/*
+    - input: photon energy [eV], threshold [eV], unknown [eV]
+    - output: unknown [no unit]
+    - called by: crossection
+*/
+    double wth = w+th;
+    if (x <= th) {
+        return 0.;
+    } else if ( (x > th) && (x < wth) ) {
+        return (x-th)/w;
+    } else if (x >= wth) {
+        return 1.;
+    } else {
+        throw std::runtime_error("error in function Ef");
+    }
+}  // end Ef
+
+
+double Photon_Field::breitwigner(double sigma_0,
+                                 double Gamma,
+                                 double DMM,
+                                 double epsPrime,
+                                 bool onProton) const {
+/*
+    - input: cross section [µbarn], width [GeV], mass [GeV/c^2]
+    - output: Breit-Wigner crossection of a resonance widh width Gamma
+    - called by: crossection
+*/
+    const double mass = onProton? 0.93827 : 0.93947;  // Gev/c^2
+    double s = mass*mass + 2.*mass*epsPrime;
+    double gam2s = Gamma*Gamma*s;
+    return sigma_0 * (s/epsPrime/epsPrime)*gam2s
+                   / ((s-DMM*DMM)*(s-DMM*DMM)+gam2s);
+}  // end breitwigner
+
+
+double Photon_Field::functs(double s, bool onProton) const {
+/*
+    - input: s [GeV^2] 
+    - output: (s-p^2)*sigma_(nucleon/gamma) [GeV^2*area]
+    - called by: sample_eps
+*/  
+    const double mass = onProton? 0.93827 : 0.93947;  // Gev/c^2
+    double factor = s - mass*mass;
+    double epsPrime = factor/2./mass;
+    double sigma_pg = crossection(epsPrime, onProton);
+    return factor*sigma_pg;
+}  // end functs
+
+
+double Photon_Field::gaussInt(double A, double B, bool onProton) const {
```

Comment:
The gauss integration should go to some general implmenetation and not be hidden inside the field